crates/nu-command/src/path/relative_to.rs

```diff
@@ -144,18 +144,68 @@ path."#
 fn relative_to(path: &Path, span: Span, args: &Arguments) -> Value {
     let lhs = expand_to_real_path(path);
     let rhs = expand_to_real_path(&args.path.item);
+
     match lhs.strip_prefix(&rhs) {
         Ok(p) => Value::string(p.to_string_lossy(), span),
-        Err(e) => Value::error(
-            ShellError::CantConvert {
-                to_type: e.to_string(),
-                from_type: "string".into(),
+        Err(e) => {
+            // On case-insensitive filesystems, try case-insensitive comparison
+            if is_case_insensitive_filesystem() {
+                if let Some(relative_path) = try_case_insensitive_strip_prefix(&lhs, &rhs) {
+                    return Value::string(relative_path.to_string_lossy(), span);
+                }
+            }
+
+            Value::error(
+                ShellError::CantConvert {
+                    to_type: e.to_string(),
+                    from_type: "string".into(),
+                    span,
+                    help: None,
+                },
                 span,
-                help: None,
-            },
-            span,
-        ),
+            )
+        }
+    }
+}
+
+/// Check if the current filesystem is typically case-insensitive
+fn is_case_insensitive_filesystem() -> bool {
+    // Windows and macOS typically have case-insensitive filesystems
+    cfg!(any(target_os = "windows", target_os = "macos"))
+}
+
+/// Try to strip prefix in a case-insensitive manner
+fn try_case_insensitive_strip_prefix(lhs: &Path, rhs: &Path) -> Option<std::path::PathBuf> {
+    let lhs_components: Vec<_> = lhs.components().collect();
+    let rhs_components: Vec<_> = rhs.components().collect();
+
+    if rhs_components.len() > lhs_components.len() {
+        return None;
+    }
+
+    for (lhs_comp, rhs_comp) in lhs_components.iter().zip(rhs_components.iter()) {
```

Comment:
I'm thinking that we don't need to collect `components` at first time.  We can just use the `Components` iterator.
```
let mut lhs_components = lhs.components();
let mut rhs_components = rhs.components();

let mut common_cnt = 0;
for (lhs_comp, rhs_comp) in lhs.components().zip(rhs.components()) {
    ...
    common_cnt += 1
}
let mut result = std::path::PathBuf::new();
for component in lhs_components.iter().skip(common_cnt) {
    result.push(component);
}
```